src/modules/silentpayments/tests_impl.h

```diff
@@ -347,11 +348,244 @@ static void test_recipient_api(void) {
+        CHECK(secp256k1_keypair_create(CTX, &taproot_keypairs[i], test->taproot_seckeys[i]));
+        taproot_keypair_ptrs[i] = &taproot_keypairs[i];
+    }
+    ret = secp256k1_silentpayments_sender_create_outputs(CTX,
+                generated_output_ptrs,
+                recipient_ptrs,
+                test->num_outputs,
+                test->outpoint_smallest,
+                test->num_taproot_inputs > 0 ? taproot_keypair_ptrs : NULL, test->num_taproot_inputs,
+                test->num_plain_inputs > 0 ? plain_seckeys : NULL, test->num_plain_inputs
+    );
+    /* If we are unable to create outputs, e.g., the input keys sum to zero, check that the
+     * expected number of recipient outputs for this test case is zero
+     */
+    if (!ret) {
+        CHECK(test->num_recipient_outputs == 0);
+        return;
+    }
+
+    match = 0;
+    for (i = 0; i < test->num_output_sets; i++) {
+        size_t n_matches = 0;
+        for (j = 0; j < test->num_outputs; j++) {
+            CHECK(secp256k1_xonly_pubkey_serialize(CTX, created_output, &generated_outputs[j]));
+            /* Loop over both lists to ensure tests don't fail due to different orderings of outputs */
+            for (k = 0; k < test->num_recipient_outputs; k++) {
+                if (secp256k1_memcmp_var(created_output, test->recipient_outputs[i][k], 32) == 0) {
+                    n_matches++;
+                    break;
+                }
+            }
+        }
+        if (n_matches == test->num_recipient_outputs) {
+            match = 1;
+            break;
+        }
+    }
+    CHECK(match);
+}
+
+void run_silentpayments_test_vector_receive(const struct bip352_test_vector *test) {
+    secp256k1_pubkey plain_pubkeys_objs[MAX_INPUTS_PER_TEST_CASE];
+    secp256k1_xonly_pubkey xonly_pubkeys_objs[MAX_INPUTS_PER_TEST_CASE];
+    secp256k1_xonly_pubkey tx_output_objs[MAX_OUTPUTS_PER_TEST_CASE];
+    secp256k1_silentpayments_found_output found_output_objs[MAX_OUTPUTS_PER_TEST_CASE];
+    secp256k1_pubkey const *plain_pubkeys[MAX_INPUTS_PER_TEST_CASE];
+    secp256k1_xonly_pubkey const *xonly_pubkeys[MAX_INPUTS_PER_TEST_CASE];
+    secp256k1_xonly_pubkey const *tx_outputs[MAX_OUTPUTS_PER_TEST_CASE];
+    secp256k1_silentpayments_found_output *found_outputs[MAX_OUTPUTS_PER_TEST_CASE];
+    unsigned char found_outputs_light_client[MAX_OUTPUTS_PER_TEST_CASE][32];
+    secp256k1_pubkey recipient_scan_pubkey;
+    secp256k1_pubkey recipient_spend_pubkey;
+    size_t i,j;
+    int match, ret;
+    size_t n_found = 0;
+    unsigned char found_output[32];
+    unsigned char found_signatures[10][64];
+    secp256k1_silentpayments_public_data public_data, public_data_index;
+    unsigned char shared_secret_lightclient[33];
+    unsigned char light_client_data[33];
+
+
+    /* prepare the inputs */
+    {
+        for (i = 0; i < test->num_plain_inputs; i++) {
+            CHECK(secp256k1_ec_pubkey_parse(CTX, &plain_pubkeys_objs[i], test->plain_pubkeys[i], 33));
```

Comment:
ef597945b2632862158a9477a74c1a525e92ffe9: to sanity check the test vectors themselves, if a matching `plain_seckeys[i]` entry exists, derive the public key to see that it matches. I can currently change a byte in the first test vector sec and pub key without breaking a test.

Reply:
Mmm, even if malform `vectors.h` by deleting the `struct bip352_test_vector {` line `make check` is fine. That take a `make clean`. Seems like a Makefile bug.

Reply:
> Mmm, even if malform vectors.h by deleting the struct bip352_test_vector { line make check is fine.

Sounds like make is not picking up that `vectors.h` changed? Will look into this.

> to sanity check the test vectors themselves, if a matching plain_seckeys[i] entry exists, derive the public key to see that it matches

This would make the tests run slower, and I don't really see the value. Since the values for the tests are static, seems sufficient to do a one-time manual check as part of testing vs make this run every time in the CI?

Reply:
> a one-time manual check

Maybe the Python script can do it?

Reply:
You can add the file here:
https://github.com/bitcoin-core/secp256k1/blob/0055b86780f2aa7272a1c307f6b9cd298584340f/Makefile.am#L237-L243 

Then CI will check that the files match:
https://github.com/bitcoin-core/secp256k1/blob/0055b86780f2aa7272a1c307f6b9cd298584340f/ci/ci.sh#L138-L147

Note in case you're wondering: There's no CMake equivalent for this. It's our own internal stuff, so there was no need to port it to CMake.

Reply:
Added.